.github/workflows/test.yml

```
@@ -81,10 +87,53 @@ jobs:
       - name: Initialize test db
         run: oca_init_test_database
       - name: Run tests
-        run: oca_run_tests
+        id: run_tests
+        run: oca_run_tests 2>&1 | tee odoo-${{matrix.id}}.log
+      - name: Upload
+        uses: actions/upload-artifact@v4
+        with:
+          name: odoo-${{matrix.id}}-log
+          path: odoo-${{matrix.id}}.log
       - uses: codecov/codecov-action@v4
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
       - name: Update .pot files
         run: oca_export_and_push_pot https://x-access-token:${{ secrets.GIT_PUSH_TOKEN }}@github.com/${{ github.repository }}
         if: ${{ matrix.makepot == 'true' && github.event_name == 'push' && github.repository_owner == 'OCA' }}
+  
+  # merge:
+  #   runs-on: ubuntu-latest
+  #   needs: test
+  #   steps:
+  #     - name: Merge Artifacts
+  #       uses: actions/upload-artifact/merge@v4
+  #       with:
+  #         name: my-amazing-merged-artifact
+  #         pattern: odoo-*
+  #         delete-merged: true
+  #         retention-days: 1
+
+  checklog:
+    needs: test
+    runs-on: ubuntu-22.04
+    name: checklog-odoo
+    steps:
+      - name: Download
+        uses: actions/download-artifact@v4
+      - name: Finding files
+        id: finding-files        
+        run: |
+          {
+            echo 'FILELIST<<EOF'
+            find . -name '*.log' -print 
+            echo EOF
+          } >> "$GITHUB_ENV"
+      - name: Check logs
+        run: |
+          pip install checklog-odoo
+          echo $FILELIST
+          {
+          for x in $FILELIST; do
```

```suggestion
          for x in $(ls *.log); do
```

You can probably do this in one step like this.